# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: 8af30c6d9bb9
# Full Hash: 8af30c6d9bb91a1ed9a5bee712252bbe1bd20d85
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 11: Use TaggedParserAtomIndex in ScriptThingVariant. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96918
# ==============================================================================

diff -r cb0e58368cd4 -r 8af30c6d9bb9 js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:28 2020 +0000
+++ b/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:36 2020 +0000
@@ -1492,17 +1492,17 @@
   }
 }
 
-static void DumpScriptThing(js::JSONPrinter& json, ScriptThingVariant& thing) {
+static void DumpScriptThing(js::JSONPrinter& json,
+                            CompilationStencil* compilationStencil,
+                            ScriptThingVariant& thing) {
   struct Matcher {
     js::JSONPrinter& json;
+    CompilationStencil* compilationStencil;
 
-    void operator()(ScriptAtom& data) {
+    void operator()(TaggedParserAtomIndex& data) {
       json.beginObject();
       json.property("type", "ScriptAtom");
-      const ParserAtom* atom = data;
-      GenericPrinter& out = json.beginStringProperty("value");
-      atom->dumpCharsNoQuote(out);
-      json.endStringProperty();
+      DumpTaggedParserAtomIndex(json, data, compilationStencil);
       json.endObject();
     }
 
@@ -1533,7 +1533,7 @@
     }
   };
 
-  Matcher m{json};
+  Matcher m{json, compilationStencil};
   thing.match(m);
 }
 
@@ -1565,7 +1565,7 @@
 
   json.beginListProperty("gcThings");
   for (auto& thing : gcThings) {
-    DumpScriptThing(json, thing);
+    DumpScriptThing(json, compilationStencil, thing);
   }
   json.endList();
 